R/prepare_matches_mapper.R

```diff
@@ -6,16 +6,24 @@
 #' @return A dataframe that contains information on multiple matches
 #' @noRd
 prepare_matches_mapper <- function(mapper, activities) {
+  stopifnot(is.logical(mapper$multi_match))
   activities <- activities |>
     select("activity_uuid_product_uuid", "activity_name", "reference_product_name", "unit")
 
-  mapper |>
-    left_join(activities, by = "activity_uuid_product_uuid") |>
-    summarise(across(everything(), ~ paste0(na.omit(unique(.)), collapse = "; ")), .by = c("country", "main_activity", "clustered")) |>
-    ungroup() |>
+  mapper_df <- mapper |>
+    left_join(activities, by = "activity_uuid_product_uuid")
+  stop_if_some_columns_have_more_than_one_unique_value(mapper_df)
+
+  non_summarised_df <- mapper_df |>
+    select(-c("activity_uuid_product_uuid", "activity_name")) |>
+    summarise(across(everything(), \(x) unique(x)), .by = c("country", "main_activity", "clustered"))
+  summarised_df <- mapper_df |>
+    select(c("country", "main_activity", "clustered", "activity_uuid_product_uuid", "activity_name")) |>
+    summarise(across(everything(), \(x) paste0(na.omit(unique(x)), collapse = "; ")), .by = c("country", "main_activity", "clustered"))
+
+  non_summarised_df |>
+    left_join(summarised_df, by = c("country", "main_activity", "clustered")) |>
+    mutate(across(everything(), \(x) ifelse(!nzchar(x), NA_character_, x))) |>
```

Comment:
This became more complicated. It's hard for me to understand what's exactly your intent but this might help:

The call to `summarize()` with `unique()` seems to do something similar to `distinct()`. So maybe something like this works for you?

```r
  # ...
  mapper_df |>
    distinct(country, main_activity, clustered) |>
    # I'm not sure about this though. Maybe this inserts the duplicates that the line above removes?
    right_join(mapper_df) |>
    #...
```

Reply:
@maurolepore  Thanks for suggesting this! 
However, after merging this code #67 from another PR, I found new errors. To resolve them, I have to refactor the code a lot. Please review it again! Thanks!